### DISCUSSION TITLE
Configure basetemp in ini config file

### DISCUSSION THREAD

**Discussion by yihuang** [pytest-dev/pytest#7876] [General] [answered]
Since we have [this issue](https://github.com/pytest-dev/pytest/issues/5802#issuecomment-705872426), in our project, we have to pass `--basetemp=/tmp/pytest` argument every time. I suggest we support setting this option in ini file.

**Comment by symonk** [comment]
I’m happy to implement this if we have agreement, I’m kind of +0, due to the nature of basetemp deleting the directory contents, I like forcing it explicitly almost like an ‘acceptance’ from the user of what it can do as a side effect.

**Comment by yihuang** [comment]
> I’m happy to implement this if we have agreement, I’m kind of +0, due to the nature of basetemp deleting the directory contents, I like forcing it explicitly almost like an ‘acceptance’ from the user of what it can do as a side effect.

Or can we make a shorter default, like "/tmp/pytest" on posix system?

**Comment by nicoddemus** [comment, marked as answer]
> Since we have this issue, in our project, we have to pass --basetemp=/tmp/pytest argument every time. I suggest we support setting this option in ini file.

You can already add this to your `pytest.ini` file:

```ini
[pytest]
addopts = --basetemp=/tmp/pytest
```

I believe this is enough to work around the problem?

**Comment by yihuang** [comment]
Another solution works for me is like this:
```
TMPDIR=/tmp pytest
```
The test directory would be `/tmp/pytest-of-{user}/pytest-{age}/`, and I can still enjoy the cleanup feature ;D

Do you know why `tempfile.gettempdir()` on mac returns such a long directory name?

EDIT: answer my own question, the default value of `TMPDIR` on mac is the `/var/folders/......`, `gettempdir` just returned that. the reason for that maybe create sort of isolation between sessions?

**Comment by yihuang** [comment]
I'm closing this one, since we have enough work-arounds.

**Comment by nicoddemus** [comment]
> EDIT: answer my own question, the default value of TMPDIR on mac is the /var/folders/......, gettempdir just returned that. the reason for that maybe create sort of isolation between sessions?

I wouldn't know 😁

**Comment by nicoddemus** [comment]
Glad we found a suitable workaround then.
